packages/beacon-node/src/network/reqresp/types.ts

```diff
@@ -99,7 +110,7 @@ const blocksResponseType: ResponseTypeGetter<SignedBeaconBlock> = (fork, version
 };
 
 export const responseSszTypeByMethod: {[K in ReqRespMethod]: ResponseTypeGetter<ResponseBodyByMethod[K]>} = {
-  [ReqRespMethod.Status]: () => ssz.phase0.Status,
+  [ReqRespMethod.Status]: (_, version) => (version === Version.V2 ? ssz.fulu.Status : ssz.phase0.Status),
```

Comment:
why do we use the version instead of `fork` (first param `_`) to determine type?

**Edit:** noticed it's same for `Metadata` is it because we use them before fork already? might wanna leave a comment here if that's the case